src/sarif_converter.ts

```diff
@@ -187,3 +203,39 @@ const getResults = (suggestions: ISarifSuggestions) => {
   }
   return output;
 };
+
+//custom string splice implementation
+function spliceSlice(str: string, index: number, count: number, add: string) {
```

Comment:
`required`: this function can use some refactoring love :-) Couple of suggestions:
- `spliceSlice` is not very clear name
- parameter `add` is consider optional (`add || ''`), but is not marked that way (`add?: string`)
- main difference between `slice` and `splice` on Array is in the in place modification; not sure if this modifies anything, therefore `splice` naming can be misleading
- I somehow believe there might be an easier way how to implement this, but it is hard to say without tests
- this logic is not trivial, therefore it should be tested